gateway/testutil.go

```diff
@@ -162,14 +166,12 @@ func (r *ReloadMachinery) Queued() bool {
 // EnsureQueued this will block until any queue happens. It will timeout after
 // 100ms
 func (r *ReloadMachinery) EnsureQueued(t *testing.T) {
-	deadline := time.NewTimer(100 * time.Millisecond)
-	defer deadline.Stop()
 	tick := time.NewTicker(time.Millisecond)
 	defer tick.Stop()
 	for {
 		select {
-		case <-deadline.C:
-			t.Fatal("Timedout waiting for reload to be queue")
+		case <-time.After(100 * time.Millisecond):
+			t.Fatal("Timedout waiting for reload to be queued")
```

Comment:
I know this are test but this will introduce memory leak, when `<-tick.C` happens resources from `<-time.After` will not get cleaned.

Reply:
Same in line `190`